receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_scraper.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filesystemscraper
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/shirou/gopsutil/disk"
+	"go.opencensus.io/trace"
+
+	"github.com/open-telemetry/opentelemetry-collector/component/componenterror"
+	"github.com/open-telemetry/opentelemetry-collector/consumer"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/pdata"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/pdatautil"
+	"github.com/open-telemetry/opentelemetry-collector/internal/data"
+	"github.com/open-telemetry/opentelemetry-collector/receiver/hostmetricsreceiver/internal"
+)
+
+// Scraper for FileSystem Metrics
+type Scraper struct {
+	config   *Config
+	consumer consumer.MetricsConsumer
+	cancel   context.CancelFunc
+}
+
+type deviceUsage struct {
+	deviceName string
+	usage      *disk.UsageStat
+}
+
+// NewFileSystemScraper creates a set of FileSystem related metrics
+func NewFileSystemScraper(ctx context.Context, cfg *Config, consumer consumer.MetricsConsumer) (*Scraper, error) {
+	return &Scraper{config: cfg, consumer: consumer}, nil
+}
+
+// Start
+func (s *Scraper) Start(ctx context.Context) error {
+	ctx, s.cancel = context.WithCancel(ctx)
+
+	go func() {
+		ticker := time.NewTicker(s.config.CollectionInterval())
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ticker.C:
+				s.scrapeMetrics(ctx)
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+
+	return nil
+}
+
+// Shutdown
+func (s *Scraper) Shutdown(ctx context.Context) error {
```

Comment:
Note the build won't succeed until https://github.com/open-telemetry/opentelemetry-collector/pull/924 is merged due to this function being renamed

Reply:
Done.